sled-agent/src/instance.rs

```diff
@@ -385,6 +433,34 @@ impl InstanceRunner {
                                 warn!(self.log, "InstanceRunner failed to send to InstanceMonitorRunner");
                             }
                         },
+                         Some(InstanceMonitorRequest { update: Error(code), tx }) => {
+                            // TODO(eliza): this is kinda jank but Propolis
+                            // currently could return the same error when the
+                            // instance has not been ensured and when it's still
+                            // coming up, which is sad. I'm not sure whether
+                            // that happens IRL or not --- investigate this.
+                            let reaction = if has_ever_seen_a_good_state {
+                                info!(
+                                    self.log,
+                                    "Error code from Propolis after instance initialized, assuming it's gone";
+                                    "error" => ?code,
+                                );
+                                self.terminate(true).await;
+                                self.publish_state_to_nexus().await;
```

Comment:
This is probably why we sent the failed state to nexus twice in my [manual testing](https://github.com/oxidecomputer/omicron/pull/6726#issuecomment-2384012654), so I think it can be removed.